components/ILIAS/Registration/src/DualOptIn/Entity/PendingRegistrationId.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace ILIAS\Registration\DualOptIn\Entity;
+
+use ILIAS\Data\UUID\Factory as UUIDFactory;
+
+class PendingRegistrationId
+{
+    private string $uuid;
+
+    public function __construct(string $uuid)
```

Comment:
Can we make the class `readonly` and `final`?

Furthermore, we should move it into a folder (and another sub-namespace) `ValueObjects`.